### PR TITLE
Made the change to the metadata to update the Data Frame Metadata grid

### PR DESCRIPTION
**Pull request by N-thony**
Fixes (partially) #7351 
@rdstern @lilyclements I have updated the metadata to be able to drop the key columns when there is no keys.
These changes concern items a) b) and c), please take a look.

### CODE REVIEW

**Comment by N-thony** [comment]
> Fixes (partially) #7351 @rdstern @lilyclements I have updated the metadata to be able to drop the key columns when there is no keys. These changes concern items a) b) and c), please take a look.

File: instat/static/InstatObject/R/data_object_R6.R [hunk 1]
```diff
@@ -2228,7 +2228,8 @@ DataSheet$set("public", "add_key", function(col_names, key_name) {
     self$append_to_variables_metadata(col_names, is_key_label, TRUE)
     if(length(private$keys) == 1) self$append_to_variables_metadata(setdiff(self$get_column_names(), col_names), is_key_label, FALSE)
     self$append_to_metadata(is_linkable, TRUE)
-    self$append_to_metadata(next_default_item(key_label, names(self$get_metadata())), paste(col_names, collapse = ","))
+    #self$append_to_metadata(next_default_item(key_label, names(self$get_metadata())), paste(col_names, collapse = ","))
```
Comment (lilyclements) [inline instat/static/InstatObject/R/data_object_R6.R:2231]: Remove this if we don't use it?

File: instat/static/InstatObject/R/instat_object_R6.R [hunk 2]
```diff
@@ -1020,7 +1020,7 @@ DataBook$set("public", "delete_dataframes", function(data_names, delete_graph_bo
 
 DataBook$set("public", "remove_link", function(link_name) {
   if(!link_name %in% names(private$.links)) stop(link_name, " not found.")
-  private$.links[[link_name]] <- NULL
+  #private$.links[[link_name]] <- NULL
```
Comment (lilyclements) [inline instat/static/InstatObject/R/instat_object_R6.R:1023]: and this?

**Review by lilyclements — COMMENTED** [review]
@N-thony this looks good. However, there is still a column called "Is_Key" in the data frame metadata that is created when a key creates, but does not remove when the key is removed.

1. Upon opening the survey data I have the following metadata
image

2. I add a key and so the metadata shows the key column

image

3. I remove the key. But we now have a variable "Is_Key" in the metadata
image

File: instat/static/InstatObject/R/instat_object_R6.R (hunk 2 above)
Comment (N-thony) [reply instat/static/InstatObject/R/instat_object_R6.R]: Thanks @lilyclements , I resolved your comments.

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@N-thony There is still an open comment from @lilyclements . If you can resolve this, then I'm happy to approve

**Comment by N-thony** [comment]
> @N-thony There is still an open comment from @lilyclements . If you can resolve this, then I'm happy to approve

@lloyddewit I think all is resolved now.

**Comment by lilyclements** [comment]
Looks good! @lloyddewit @rdstern this is ready for you to review

**Review by rdstern — APPROVED** [review]
I'm not sure this is quite right yet, but it is much better than before, so I'm approving.  It looks perfect anyway.
a) I created key automatically in survey data through the Row Numbers/Names
b) Then key 1 from Village + Field.  This replaces the row in the metadata.  Good.  Could it also delete key?  I don't think it does - and that might be all that's needed.  It could even report in the Output Window that key1 defined, key delewted?

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**